sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -59,7 +60,7 @@ def __init__(
 
         super(MsalCredential, self).__init__()
 
-    def __enter__(self) -> "MsalCredential":
+    def __enter__(self) -> "Self":
```

Comment:
Don't believe we need quotes (forward references) here since Self is defined through the import.

```suggestion
    def __enter__(self) -> Self:
```